### PR TITLE
feat: env management commands

### LINKED ISSUE #18: Permit Environment Management Commands
We want to create a set of commands that will help users manage their Permit environments. The following is the skeleton of the commands required in this issue:

- `permit env select` - selecting a different environment and storing the relevant key. This should incorporate with the following issue and use the same components: https://github.com/permitio/permit-cli/issues/13
- `permit env copy` - will allow users to copy their existing logged-in environment to a new environment and change their active environment to it
- `permit env member` will allow users to add more members to their environment with proper roles

Acceptance criteria:
- All the above flows are impkemented
- Unit test coverage of 90%
- e2e test of the basic flow of each

### PR DESCRIPTION
**Pull request by 35C4n0r**
Closes #18 
/claim #18

### CODE REVIEW

**Comment by gemanor** [comment]
@35C4n0r the lint has some errors..

**Comment by gemanor** [comment]
Hey, some functional comments:
1. When running `env select` commands, and there's only one workspace or one project, it should skip the selection. For now, even if one project/environment is presented, it let's the user choose between them. Please ensure if there's only one element in a select step, it gets skipped.
2. In `copy` command, it is hard to get the env id. We should keep it in the same way we did with other commands. Using the select element. Please use the same SelectEnvironment that allows the user to select the environment they want to copy from a select element. Then, just ask for the name of the new environment and copy it.
3. In general, this `ink-form` makes things complex. The default should be copied to a new environment, and the form is just a wizard, the same as we do with other commands. Only in case that `--existing` argument is passed will the existing ID be asked for and copied.
4. Also, in the member command, please just use a wizard as we did in the login and such. The form is making the experience complex. If someone want's a form, then they can use it in the UI.

**Comment by gemanor** [comment]
Please merge from main

**Comment by gemanor** [comment]
Hey, @35C4n0r, the `copy` command does not work for me. Can you please test that?

**Comment by 35C4n0r** [comment]
@gemanor, pushed a change that fixes the issue, It now handles the case when `--env-name` is not passed as a cli option

**Comment by gemanor** [comment]
@35C4n0r, the scope is redundnat (just use the default), and the description/strategy should be also in the wizard, not only as variables

**Comment by 35C4n0r** [comment]
@gemanor done!

**Comment by gemanor** [comment]
The tests are failing now. Also, the success message is missing from the copy command.
Please make sure it is all passed and that you're running sanity tests on all the commands before asking review.

**Comment by gemanor** [comment]
There are also many linting warning. Please ensure you're fixing the useEffect dependencies problems. Just use the standard practices like useCallback/useMemo and pure functions to avoid such.

**Comment by 35C4n0r** [comment]
As for the warnings, if I don't want something to be the dependency of a useEffect, why should I add it in the dependency array?

**Comment by Abiji-2020** [comment]
> As for the warnings, if I don't want something to be the dependency of a useEffect, why should I add it in the dependency array?

instead of adding it as a dependency array, we can use `useCallback` or `useMemo` to overcome it, it was suggested by gemanor in early PR of mine.

File: source/hooks/useApiKeyApi.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { useMemo } from 'react';
+import { apiCall } from '../lib/api.js';
+
+export interface ApiKeyScope {
+	organization_id: string;
+	project_id: string | null;
+	environment_id: string | null;
+}
+
+export const useApiKeyApi = () => {
```
Comment (gemanor) [inline source/hooks/useApiKeyApi.ts:10]: Could be use for all the API Key level errors etc.

**Review by gemanor — CHANGES_REQUESTED** [review]
Great job! Appreciate the effort!

Please address the comments and attach a sanity test video for the following flows after addressing them)

- Environment copy
  - To new one with a wizard
  - To a new one with variables provided (all provided, and partially provided - see CR)
  - To an existing environment that exists
  - To an existing environment that does not exist (fail)
- Invite member
  - With organization-level key
  - With project-level key
  - With wrong key
  - With environment-level key
  - With provided project/environment variables (see CR)
  - With an email that is already a member
- Environment select
  - With all optional values as variables
- Login flow on Windows (you said something about the port?)

File: source/commands/env/member.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+import zod from 'zod';
+import { type infer as zInfer } from 'zod';
+import { ApiKeyScope, useApiKeyApi } from '../../hooks/useApiKeyApi.js';
+import SelectInput from 'ink-select-input';
+import { useMemberApi } from '../../hooks/useMemberApi.js';
+import EnvironmentSelection, {
+	ActiveState,
+} from '../../components/EnvironmentSelection.js';
+
+export const options = zod.object({
```
[the remark this answers is not in the document]
Comment (35C4n0r) [reply source/commands/env/member.tsx:17]: I don't think we need `project` here, we get that either way from the apiKey, We'll just add environment.

**Comment by gemanor** [comment]
@35C4n0r any updates on that?

**Review by gemanor — APPROVED** [review]
Good job, thanks!

**Pull request merged**